core/domain/state_domain.py

```diff
@@ -3137,6 +3149,18 @@ def convert_html_fields_in_state(
         if interaction_id is None:
             return state_dict
 
+        # TODO(#11950): Drop the following 'if' clause once all snapshots have
+        # been migrated. This is currently causing issues in migrating old
+        # snapshots to schema v34 because MathExpressionInput was still around
+        # at the time. It is conceptually OK to ignore customization args here
+        # because the MathExpressionInput has no customization arg fields.
+        if interaction_id == 'MathExpressionInput':
+            if state_dict['interaction']['solution']:
+                state_dict['interaction']['solution']['explanation']['html'] = (
+                    conversion_fn(state_dict['interaction']['solution'][
+                        'explanation']['html']))
```

Comment:
So this function is called in a schema migration before the schema migration to the new math interactions?

Reply:
Yup, this is correct. It was called in schema v33 and the math interactions migration happened in the conversion to schema v34.

We probably need to figure out a way to deal with this sort of thing in the future...

Reply:
I think we need to more explicitly version interactions & their interaction with particular schema versions. FWIW I plan to spec this out as part of the broader Android versioning work since it's essential to ensure changes to interactions don't break already-deployed versions of the Android app.

Reply:
Yes, I agree. I was surprised that we ran into this and I think future overhauls to the migration system would need to take it into account. Thanks for taking a look into this!

Reply:
Yeah, sounds good!